backend/plonk/bn254/solidity.go

```diff
@@ -540,18 +538,37 @@ contract PlonkVerifier {
       
       return pi;
     }
+  
+  function check_inputs_size(uint256[] memory public_inputs)
```

Comment:
Looks good to me

Reply:
@DavePearce no there is no bound on the number of public inputs, given a proof there is one exact set of public inputs that makes the tuple (proof, public_inputs) valid and vice versa